File: kotlin/lib/src/main/kotlin/EndpointAutoconfig.kt
// this file is @generated
package com.svix.kotlin

import com.svix.kotlin.models.AutoConfigOut
import com.svix.kotlin.models.RotateSubscriptionIn2
import okhttp3.Headers

data class EndpointAutoconfigCreateOptions(val idempotencyKey: String? = null)

data class EndpointAutoconfigRotateOptions(val idempotencyKey: String? = null)

class EndpointAutoconfig(private val client: SvixHttpClient) {
    /** Create an AutoConfig subscription. */
    suspend fun create(
        appId: String,
        options: EndpointAutoconfigCreateOptions = EndpointAutoconfigCreateOptions(),
    ): AutoConfigOut {
        val url = client.newUrlBuilder().encodedPath("/api/v1/app/$appId/autoconfig")
        val headers = Headers.Builder()
        options.idempotencyKey?.let { headers.add("idempotency-key", it) }
        return client.executeRequest<Any, AutoConfigOut>(
            "POST",
            url.build(),
            headers = headers.build(),
        )
    }

    /** Rotate the auth token and signing secret for an AutoConfig subscription. */
    suspend fun rotate(
        appId: String,
        autoconfigId: String,
        rotateSubscriptionIn2: RotateSubscriptionIn2,
        options: EndpointAutoconfigRotateOptions = EndpointAutoconfigRotateOptions(),
    ): AutoConfigOut {
        val url =
            client.newUrlBuilder().encodedPath("/api/v1/app/$appId/autoconfig/$autoconfigId/rotate")
        val headers = Headers.Builder()
        options.idempotencyKey?.let { headers.add("idempotency-key", it) }

        return client.executeRequest<RotateSubscriptionIn2, AutoConfigOut>(
            "POST",
            url.build(),
            headers = headers.build(),
            reqBody = rotateSubscriptionIn2,
        )
    }
}

File: ruby/lib/svix/api/endpoint_autoconfig.rb
# frozen_string_literal: true
# This file is @generated

require "net/http"

module Svix
  class EndpointAutoconfig
    def initialize(client)
      @client = client
    end

    def create(app_id, options = {})
      options = options.transform_keys(&:to_s)
      res = @client.execute_request(
        "POST",
        "/api/v1/app/#{app_id}/autoconfig",
        headers: {
          "idempotency-key" => options["idempotency-key"]
        }
      )
      AutoConfigOut.deserialize(res)
    end

    def rotate(app_id, autoconfig_id, rotate_subscription_in2, options = {})
      options = options.transform_keys(&:to_s)
      res = @client.execute_request(
        "POST",
        "/api/v1/app/#{app_id}/autoconfig/#{autoconfig_id}/rotate",
        headers: {
          "idempotency-key" => options["idempotency-key"]
        },
        body: rotate_subscription_in2
      )
      AutoConfigOut.deserialize(res)
    end

  end
end
